emulator/CMakeLists.txt

```
@@ -0,0 +1,41 @@
+foreach (arch IN LISTS ENABLED_ARCHITECTURES)
+
+    add_executable(emulator_${arch}
+        "${CMAKE_BINARY_DIR}/riscv_model_${arch}.c"
+        "riscv_config.h"
+        "riscv_platform.c"
+        "riscv_platform.h"
+        "riscv_platform_impl.c"
+        "riscv_platform_impl.h"
+        "riscv_prelude.c"
+        "riscv_prelude.h"
+        "riscv_sail.h"
+        "riscv_sim.c"
+        "riscv_softfloat.c"
+        "riscv_softfloat.h"
+    )
+
+    add_dependencies(emulator_${arch} generated_model_${arch})
+
+    target_link_libraries(emulator_${arch}
+        PRIVATE softfloat sail_runtime gmp z
```

```suggestion
        PRIVATE softfloat sail_runtime GMP::GMP ZLIB::ZLIB
```
Use the imported targets from the find modules?